Include/internal/pycore_ceval.h

```diff
@@ -232,7 +232,7 @@ static inline void _Py_LeaveRecursiveCall(void)  {
 
 extern struct _PyInterpreterFrame* _PyEval_GetFrame(void);
 
-PyAPI_FUNC(PyObject *)_Py_MakeCoro(PyFunctionObject *func);
+PyAPI_FUNC(PyGenObject *)_Py_MakeCoro(PyFunctionObject *func);
```

Comment:
We generally have a convention that there is a correspondence between the C type and the Python class.
e.g. if `PyXXX_Check[Exact](obj)` is true, then it is safe to cast `(PyXXXObject *)obj` and vice-versa.
This sort of breaks that for coroutines and async generators.

So, I think it best to leave this as `PyObject *`.  Does that make sense to you?